types/lodash/index.d.ts

```diff
@@ -17729,6 +18432,110 @@ declare namespace _ {
         ): LoDashExplicitObjectWrapper<TResult>;
     }
 
+    //_.updateWith
+    interface LoDashStatic {
+        /**
+         * This method is like `_.update` except that it accepts `customizer` which is
+         * invoked to produce the objects of `path`.  If `customizer` returns `undefined`
+         * path creation is handled by the method instead. The `customizer` is invoked
+         * with three arguments: (nsValue, key, nsObject).
+         *
+         * **Note:** This method mutates `object`.
+         *
+         * @static
+         * @memberOf _
+         * @since 4.6.0
+         * @category Object
+         * @param {Object} object The object to modify.
+         * @param {Array|string} path The path of the property to set.
+         * @param {Function} updater The function to produce the updated value.
+         * @param {Function} [customizer] The function to customize assigned values.
+         * @returns {Object} Returns `object`.
+         * @example
+         *
+         * var object = {};
+         *
+         * _.updateWith(object, '[0][1]', _.constant('a'), Object);
+         * // => { '0': { '1': 'a' } }
+         */
+        updateWith<TResult>(
+            object: Object,
+            path: Many<StringRepresentable>,
+            updater: Function,
+            customizer?: SetWithCustomizer<Object>
+        ): TResult;
+
+        /**
+         * @see _.updateWith
+         */
+        updateWith<U extends Function, TResult>(
```

Comment:
But in the previous overload, the updater is a `Function`, so the user can already specify any type of updater.

Generic types should always be constraining something (e.g. forcing 2 parameters to have the same type, or forcing the return value to have the same type as a parameter). If a generic type is only used once, it's not constraining anything and should be removed.